compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs

```diff
@@ -163,14 +163,26 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
         // Ensure `T: Sized` and `T: WF` obligations come last. This lets us display diagnostics
         // with more relevant type information and hide redundant E0282 errors.
         errors.sort_by_key(|e| match e.obligation.predicate.kind().skip_binder() {
+            ty::PredicateKind::Subtype(_)
+                if e.obligation.cause.span.macro_backtrace().next().is_some_and(
+                    |trace| match trace.kind {
+                        ExpnKind::Macro(_, name) => {
+                            name.as_str().rsplit("::").next() == Some("format_args_nl")
+                        }
+                        _ => false,
+                    },
+                ) =>
+            {
+                (-1, e.obligation.cause.span.index())
```

Comment:
you could use an enum with `PartialOrd` implemented on it to give these ordering precedences a name, or bump everything +1.